### ISSUE TITLE
Remove "integration-tests" directory

### ISSUE THREAD

**Issue by robertsami** [yugabyte/yugabyte-db#6853]
In general, we prefer to write tests which instrument yugabyte at a higher level. Most of our tests achieve this by using {External}MiniCluster. For some reason, we have a special `integration-tests` directory and some other tests which live next to the functionality they're testing in different directories, all of which are "integration tests" (for some definition of the term), and only some of which have an "-itest" prefix.

This creates confusion for newcomers to the codebase. Where should new tests go? Should they be integration tests? Should they go in this directory if so? Should they have the "-itest" prefix? If we have an "integration-tests" directory, do we prefer to write unit tests in general?

To remove confusion related to this inconsistency, and to clearly communicate that we prefer integration style tests, I propose that we remove this directory and move tests closer to the functionality they're testing.

**Comment by robertsami** [comment]
Note -- i would also propose we remove the `-itest` prefix from all test files

**Comment by robertsami** [comment]
@svarnau @mbautin is there a way to make these naming changes without losing history in detective?

**Comment by svarnau** [comment]
Not that I know of. If the test class/name changes, it shows up in reporting as a different test. But we don't tend to look at history for more than a few weeks of time, so the disconnect does not last all that long.

**Comment by robertsami** [comment]
@bmatican should this be in the "Technical investments" project instead of "kind/failing-test"? since this isnt really a failing test, more of a refactor/tech debt

**Comment by robertsami** [comment]
> Not that I know of. If the test class/name changes, it shows up in reporting as a different test. But we don't tend to look at history for more than a few weeks of time, so the disconnect does not last all that long.

fair point. might still be worried about backports though, where the history remains relevant for longer
